### PR TITLE
Clean up and test

### PR DESCRIPTION
**Pull request by cashmonger**
There is still more to do, but everything I tested worked, so that's good.

### CODE REVIEW

File: pxfish/pyfish.py [hunk 1]
```diff
@@ -158,97 +160,130 @@ def open_aquarium_session():
...
+        with open(os.path.join(directory, file_name)) as f:
              read_file = f.read()
 
         new_code = aq.Code.new(
-                 name=file_name[:-3], # Code Object Name 'protocol, library, cost_model, etc.'
-                 parent_id=definitions['id'], # OperationType or Library id 
-                 parent_class=definitions['parent_class'],   # 'OperationType' or 'Library'
-                 user_id=definitions['user_id'], # User ID from Code object 
-                 content=read_file # Contents of file  
+                 name=file_name[0:-3],
```
Comment (bjkeller) [inline pxfish/pyfish.py:247]: Use `os.path.splitext` for this instead. See https://docs.python.org/3/library/os.path.html#os.path.splitext

File: pxfish/pyfish.py [hunk 2]
```diff
@@ -158,97 +160,130 @@ def open_aquarium_session():
...
-    # for an operation_type: pyfish.py push <directory_name> -f <folder_name> -o <operation_type_name>
-    # pyfish.py push MyDirectory -f "Hydra Husbandry" -o "Husbandry Operation Type Name" 
     args = parser.parse_args()
     
+    aq = open_aquarium_session()
+
+    path = os.path.normpath(args.directory)
+    makedirectory(path)
+
     if args.library:
```
Comment (bjkeller) [inline pxfish/pyfish.py:273]: These should only happen if the action is pull, right?

File: pxfish/pyfish.py [hunk 3]
```diff
@@ -158,97 +160,130 @@ def open_aquarium_session():
...
-    parser = argparse.ArgumentParser()
-    parser.add_argument("action", choices=["push", "pull"], help="indicate if you want to push files or pull them.")  
-    # If you want to pull all Files, type pull <directory_name>
-    parser.add_argument("directory", help="the directory to which pulled files should be written")
-    parser.add_argument("-f", "--folder", help="the folder on Aquarium where your operation_type/library is located")
-    parser.add_argument("-l", "--library", help="the library you want to pull")
-    parser.add_argument("-o", "--operation_type", help="the operation_type you want to pull")
+    parser = argparse.ArgumentParser(description="Pull or Push files from/to Aquarium")
+    parser.add_argument("action", choices=["push", "pull"], help="Indicate if you want to push files or pull them.")  
+    parser.add_argument("-d", "--directory", required=True, help="The name of the directory where the files should be written. If the directory does not already exist, it will be created")
```
[the remark this answers is not in the document]
Comment (bjkeller) [reply pxfish/pyfish.py:259]: https://docs.python.org/3/library/os.html#os.getcwd

**Comment by cashmonger** [comment]
I'm not sure what you mean by point 1 -- that it looks like it always pulls? You can also push -- or, at least I was able to.

**Comment by bjkeller** [comment]
you just jump right into pulling without checking the action is pull

**Comment by bjkeller** [comment]
https://github.com/klavinslab/ParroPyTriFish/blob/4c98fd4263f733310a4dae54fdb2bd932a486a09/pxfish/pyfish.py#L273

File: pxfish/pyfish.py [hunk 4]
```diff
@@ -256,34 +257,41 @@ def push(aq, directory, files_to_write):
...
     parser.add_argument("-c", "--category", help="The Aquarium category where the OperationType or Library is located")
 
     group = parser.add_mutually_exclusive_group()
     group.add_argument("-l", "--library", help="the Library you want to pull")
     group.add_argument("-o", "--operation_type", help="the Operation Type you want to pull")
-    
+   
     args = parser.parse_args()
     
+    if not args.directory:
```
Comment (bjkeller) [inline pxfish/pyfish.py:269]: see note about default above

File: pxfish/pyfish.py [hunk 5]
```diff
@@ -256,34 +257,41 @@ def push(aq, directory, files_to_write):
...
+    if args.action == 'pull':
+        if args.library: 
+            get_library(aq, path, args.category, args.library) 
+        elif args.operation_type:
+            get_operation_type(aq, path, args.category, args.operation_type)
+        elif args.category: 
+            get_category(aq, path, args.category)
         else:
+            get_all_optypes_and_libraries(aq, path)
+    elif args.action == 'push':
```
Comment (bjkeller) [inline pxfish/pyfish.py:286]: you could do

```python
if args.action != 'push':
    logging.warning("You must indicate whether you would like to 'push' or 'pull'")
    return
```

and then rest would be for push, but follow similar pattern to code for pull

File: pxfish/pyfish.py [hunk 6]
```diff
@@ -71,23 +71,27 @@ def field_type_list(field_types, role):
...
-    #ot_ser["code_name"] = operation_type.protocol.name
     ot_ser["parent_class"] = "OperationType"
     ot_ser["category"] = operation_type.category
     ot_ser["inputs"] = field_type_list(operation_type.field_types, 'input')
     ot_ser["outputs"] = field_type_list(operation_type.field_types, 'output')
     ot_ser["on_the_fly"] = operation_type.on_the_fly
-    ot_ser["user_id"] = operation_type.protocol.user_id
-    
+
+    if operation_type.protocol:
```
[the remark this answers is not in the document]
Comment (bjkeller) [reply pxfish/pyfish.py:91]: I think the best way to deal with this is to move the conditional to the caller.

[the remark this answers is not in the document]
Comment (cashmonger) [reply pxfish/pyfish.py:91]: I'm going to take this out -- I updated my DB and this error is no longer happening -- I think it was an artifact of some of my experiments with pushing.

**Pull request merged**